### PR TITLE
Add event waiter helper to E2E tests. Fix onchain-logic E2E tests

### PR DESCRIPTION
**Pull request by nguyer**
Signed-off-by: Nicko Guyer <nicko.guyer@kaleido.io>

### CODE REVIEW

File: test/e2e/ethereum_contract_test.go [hunk 1]
```diff
@@ -288,23 +301,27 @@ func (suite *EthereumContractTestSuite) TestFFIInvokeMethod() {
...
+		},
+		"subscription": sub.ID.String(),
+	}
 
-	events := GetContractEvents(suite.T(), suite.testState.client1, suite.testState.startTime)
-	assert.Equal(suite.T(), 1, len(events))
-	assert.Equal(suite.T(), "Changed", events[0].Name)
-	assert.Equal(suite.T(), "42", events[0].Outputs.GetString("_value"))
-	assert.Equal(suite.T(), suite.ethIdentity, events[0].Outputs.GetString("_from"))
+	event := waitForChangeEvent(suite.T(), suite.testState.client1, changes1, match)
```
Comment (awrichar) [inline test/e2e/ethereum_contract_test.go:325]: Does this only validate the `changes1` websocket for database events? Shouldn't we drain the `received1` websocket as well (as that's the more important user-facing one for E2E purposes)?

Comment (nguyer) [reply test/e2e/ethereum_contract_test.go:325]: Do you think we should wait for both here, or just look at `received1`?

Comment (nguyer) [reply test/e2e/ethereum_contract_test.go]: Okay, updated this PR to look at the deliveryEvents

File: test/e2e/ethereum_contract_test.go [hunk 2]
```diff
@@ -161,7 +162,7 @@ type EthereumContractTestSuite struct {
 	suite.Suite
 	testState       *testState
 	contractAddress string
-	interfaceID     string
+	contractID      string
```
Comment (awrichar) [inline test/e2e/ethereum_contract_test.go:165]: Why rename this back to contractID? Not a big deal, but it is the ID of an FFI / Contract Interface.

Comment (nguyer) [reply test/e2e/ethereum_contract_test.go:165]: This was probably an unintended change while merging/rebasing branches. I will change it back.

File: test/e2e/e2e_test.go [hunk 3]
```diff
@@ -283,3 +284,89 @@ func wsReader(t *testing.T, conn *websocket.Conn) (chan *fftypes.EventDelivery,
...
+func waitForEventDelivery(t *testing.T, client *resty.Client, c chan *fftypes.EventDelivery, match map[string]interface{}) map[string]interface{} {
+	for {
+		eventDelivery := <-c
+		var event interface{}
+		var err error
+		switch eventDelivery.Type {
+		case fftypes.EventTypeContractEvent:
+			event, err = GetContractEvent(t, client, eventDelivery.Event.Reference.String())
+		default:
+			event, err = GetEvent(t, client, eventDelivery.Event.Reference.String())
```
[the remark this answers is not in the document]
Comment (nguyer) [reply test/e2e/e2e_test.go:318]: Thanks. I wondered about that. I will fix this.

File: test/e2e/restclient.go [hunk 4]
```diff
@@ -522,9 +524,9 @@ func CreateFFI(t *testing.T, client *resty.Client, ffi *fftypes.FFI) (interface{
 	return res, err
 }
 
-func InvokeFFIMethod(t *testing.T, client *resty.Client, interfaceID, methodName string, invokeContractRequest *fftypes.InvokeContractRequest) (interface{}, error) {
+func InvokeFFIMethod(t *testing.T, client *resty.Client, contractID, methodName string, invokeContractRequest *fftypes.InvokeContractRequest) (interface{}, error) {
```
Comment (awrichar) [inline test/e2e/restclient.go:527]: Still a handful of places where interfaceID got changed back to contractID...

**Review by awrichar — APPROVED** [review]

**Pull request merged**